src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -547,6 +675,38 @@ const char *RcpCapsDiag::CategoryToString(Category aCategory)
     return (aCategory < OT_ARRAY_LENGTH(kCategoryStrings)) ? kCategoryStrings[aCategory] : "invalid";
 }
 
+const char *RcpCapsDiag::SupportToString(bool aSupport) { return aSupport ? "OK" : "NotSupported"; }
+
+const char *RcpCapsDiag::RadioCapbilityToString(uint32_t aCapability)
+{
+    static const char *const kCapbilityStrings[] = {
+        "RADIO_CAPS_ACK_TIMEOUT",      // (1 << 0) OT_RADIO_CAPS_ACK_TIMEOUT
+        "RADIO_CAPS_ENERGY_SCAN",      // (1 << 1) OT_RADIO_CAPS_ENERGY_SCAN
+        "RADIO_CAPS_TRANSMIT_RETRIES", // (1 << 2) OT_RADIO_CAPS_TRANSMIT_RETRIES
+        "RADIO_CAPS_CSMA_BACKOFF",     // (1 << 3) OT_RADIO_CAPS_CSMA_BACKOFF
+        "RADIO_CAPS_SLEEP_TO_TX",      // (1 << 4) OT_RADIO_CAPS_SLEEP_TO_TX
+        "RADIO_CAPS_TRANSMIT_SEC",     // (1 << 5) OT_RADIO_CAPS_TRANSMIT_SEC
+        "RADIO_CAPS_TRANSMIT_TIMING",  // (1 << 6) OT_RADIO_CAPS_TRANSMIT_TIMING
+        "RADIO_CAPS_RECEIVE_TIMING",   // (1 << 7) OT_RADIO_CAPS_RECEIVE_TIMING
+        "RADIO_CAPS_RX_ON_WHEN_IDLE",  // (1 << 8) OT_RADIO_CAPS_RX_ON_WHEN_IDLE
+    };
+
+    static_assert(OT_RADIO_CAPS_ACK_TIMEOUT == 1 << 0, "OT_RADIO_CAPS_ACK_TIMEOUT value is incorrect");
+    static_assert(OT_RADIO_CAPS_ENERGY_SCAN == 1 << 1, "OT_RADIO_CAPS_ENERGY_SCAN value is incorrect");
+    static_assert(OT_RADIO_CAPS_TRANSMIT_RETRIES == 1 << 2, "OT_RADIO_CAPS_TRANSMIT_RETRIES value is incorrect");
+    static_assert(OT_RADIO_CAPS_CSMA_BACKOFF == 1 << 3, "OT_RADIO_CAPS_CSMA_BACKOFF value is incorrect");
+    static_assert(OT_RADIO_CAPS_SLEEP_TO_TX == 1 << 4, "OT_RADIO_CAPS_SLEEP_TO_TX value is incorrect");
+    static_assert(OT_RADIO_CAPS_TRANSMIT_SEC == 1 << 5, "OT_RADIO_CAPS_TRANSMIT_SEC value is incorrect");
+    static_assert(OT_RADIO_CAPS_TRANSMIT_TIMING == 1 << 6, "OT_RADIO_CAPS_TRANSMIT_TIMING value is incorrect");
+    static_assert(OT_RADIO_CAPS_RECEIVE_TIMING == 1 << 7, "OT_RADIO_CAPS_RECEIVE_TIMING value is incorrect");
+    static_assert(OT_RADIO_CAPS_RX_ON_WHEN_IDLE == 1 << 8, "OT_RADIO_CAPS_RX_ON_WHEN_IDLE value is incorrect");
+
+    int32_t pos = static_cast<int32_t>(ffs(static_cast<int>(aCapability))) - 1;
```

Comment:
Suggest maybe not using the `ffs()` here (avoid relying on library support for this). 
We can consider implementing our own version of such a function?

Reply:
How about something like this:
```c++
    const char *string = "invalid;"
    uint16_t   index   = 0;

    for (; !(aCapability & 0x1); aCapability >>= 1, index++)
    {
        VerifyOrExit(index < OT_ARRAY_LENGTH(kCapbilityStrings));
    }

    string = kCapbilityStrings[index];

exit:
    return string;
```

Reply:
Updated